src/interface/file.rs

```diff
@@ -20,30 +20,8 @@ use std::ffi::c_void;
 use std::convert::TryInto;
 use crate::interface::errnos::{Errno, syscall_error};
 
-static OPEN_FILES: RustLazyGlobal<Arc<DashSet<String>>> = RustLazyGlobal::new(|| Arc::new(DashSet::new()));
-
-pub fn listfiles() -> Vec<String> {
-    let paths = fs::read_dir(&RustPath::new(
-        &env::current_dir().unwrap())).unwrap();
-      
-    let names =
-    paths.filter_map(|entry| {
-      entry.ok().and_then(|e|
-        e.path().file_name()
-        .and_then(|n| n.to_str().map(|s| String::from(s)))
-      )
-    }).collect::<Vec<String>>();
-
-    return names;
-}
 
 pub fn removefile(filename: String) -> std::io::Result<()> {
-    let openfiles = &OPEN_FILES;
-
-    if openfiles.contains(&filename) {
-        panic!("FileInUse");
-    }
-
```

Comment:
I'm not certain this isn't needed.  On Linux this is probably okay.  On Windows, I think this will error since you can't remove an open file.  Certainly, you could do something better then get the whole file list.

From what I see, we only use removefile internally, for our own use.  So are you able to just kill the function altogether?

Reply:
That make sense. We'll need to audit this code further in the upcoming PRs. For now the management we do in RustPOSIX regarding linkcount should be sane.